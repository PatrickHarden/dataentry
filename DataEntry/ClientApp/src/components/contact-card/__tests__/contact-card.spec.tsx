
import React from 'react';
import { shallow } from 'enzyme';

import ContactCard from '../contact-card';

describe('components', () => {

  describe('contact-card', () => {

    const contact = {
        contactId: "3c113c1f-78df-4886-8f95-b65f6f8c7eec",
        email: "jane.smith@cbre.com",
        firstName: "Jane",
        lastName: "Smith",
        license: "123344zzz444",
        location: "Dallas, TX",
        phone: "(202) 345-5678"
    }
    
    const deleteCard = () => {
        console.log('delete')
    }
    
    const openModal = () => {
        console.log('open modal')
    }
    // snapshot render
    it('renders according to snapshot', () => {
        expect(shallow( <ContactCard contact={contact} index={0} deleteContact={deleteCard} openModal={openModal} />)).toMatchSnapshot();
    });

  });
});