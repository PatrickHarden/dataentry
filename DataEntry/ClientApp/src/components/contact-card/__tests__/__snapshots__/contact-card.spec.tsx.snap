// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components contact-card renders according to snapshot 1`] = `
<styled.div>
  <div
    className="contactLeftCol"
  >
    <ImageAvatar
      allowUpload={false}
      avatarSize="65px"
      imageURL=""
      placeholder="J S"
      placeholderSize="20px"
    />
  </div>
  <div
    className="contactRightCol"
  >
    <span
      onClick={[Function]}
    >
      <img
        src="deleteIcon.png"
      />
    </span>
    <styled.p
      id="contactNameWrapper"
    >
      Jane
       
      Smith
    </styled.p>
    <styled.p
      id="contactLocationWrapper"
    >
      Dallas, TX
    </styled.p>
    <p
      id="emailWrapper"
    >
      <a
        href="mailto: jane.smith@cbre.com"
      >
        <img
          src="icon-email.png"
        />
        jane.smith@cbre.com
      </a>
    </p>
    <p
      id="phoneWrapper"
    >
      <a
        href="tel: (202) 345-5678"
      >
        <img
          src="icon-phone.png"
        />
        (202) 345-5678
      </a>
    </p>
  </div>
</styled.div>
`;
